Automattic/wp-calypso
# Stepper: Fix back button on use my domain step

Fixes: https://github.com/Automattic/wp-calypso/issues/94234

## Proposed Changes

* Updates the back button navigation for the use my domain step

## Why are these changes being made?


* Allow users to select a choose a different domain instead of navigating back to domains and then navigating to use my domain step again.

## Testing Instructions



* Checkout branch or use live link
* Navigate to `/setup/onboarding` and click the `Use a domain I own` link on the side bar.
* Continue through the flow until you reach the checkout and verify that the correct domain cart items are added
* Test the back navigation
* Check for regressions on `/start/` flow.

## Pre-merge Checklist
